### ISSUE TITLE
Ifpack2:  clang linker error

### ISSUE THREAD

**Issue by dridzal** [trilinos/Trilinos#13076]
## Bug Report
@trilinos/tpetra @trilinos/ifpack2 

With MacOS Sonoma, Intel hardware, clang, I'm getting a linker error between Ifpack2 and Tpetra as of 8-Jun-2024.  I am not getting this error with a version of Trilinos from early April.

### Description
Error:

```
ld: Undefined symbols:
  Teuchos::RCP<Tpetra::CrsGraph<int, long long, Tpetra::KokkosCompat::KokkosDeviceWrapperNode<Kokkos::Serial, Kokkos::HostSpace>>> Tpetra::getBlockCrsGraph<double, int, long long, Tpetra::KokkosCompat::KokkosDeviceWrapperNode<Kokkos::Serial, Kokkos::HostSpace>>(Tpetra::CrsMatrix<double, int, long long, Tpetra::KokkosCompat::KokkosDeviceWrapperNode<Kokkos::Serial, Kokkos::HostSpace>> const&, int const&, bool), referenced from:
      Ifpack2::BlockRelaxation<Tpetra::RowMatrix<double, int, long long, Tpetra::KokkosCompat::KokkosDeviceWrapperNode<Kokkos::Serial, Kokkos::HostSpace>>, Ifpack2::Container<Tpetra::RowMatrix<double, int, long long, Tpetra::KokkosCompat::KokkosDeviceWrapperNode<Kokkos::Serial, Kokkos::HostSpace>>>>::initialize() in Ifpack2_BlockRelaxation_DOUBLE_INT_LONG_LONG_SERIAL.cpp.o
      Ifpack2::BlockTriDiContainer<Tpetra::RowMatrix<double, int, long long, Tpetra::KokkosCompat::KokkosDeviceWrapperNode<Kokkos::Serial, Kokkos::HostSpace>>, Ifpack2::BlockTriDiContainerDetails::ImplSimdTag>::initialize() in Ifpack2_BlockTriDiContainer_DOUBLE_INT_LONG_LONG_SERIAL.cpp.o
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

### Steps to Reproduce
1. SHA1:  latest master as of the filing of the issue, ee6599d413f7ef5786a5ad1f719bbfe3a4d22300.
2. Configure script:
```
  -D CMAKE_BUILD_TYPE:STRING=RELEASE \
  -D TPL_ENABLE_MPI:BOOL=ON \
  -D Trilinos_ENABLE_OpenMP:BOOL=OFF \
  -D BUILD_SHARED_LIBS:BOOL=ON \
  -D Trilinos_ENABLE_EXPLICIT_INSTANTIATION:BOOL=ON \
  -D TPL_ENABLE_Boost=OFF \
  -D Trilinos_ENABLE_MueLu:BOOL=ON \
  -D Trilinos_ENABLE_Panzer:BOOL=ON \
  -D Trilinos_ENABLE_Compadre:BOOL=ON \
  -D Trilinos_ENABLE_ROL:BOOL=ON \
  -D Trilinos_ENABLE_SEACAS:BOOL=ON \
  -D TPL_ENABLE_Netcdf=ON \
  -D TPL_Netcdf_PARALLEL:BOOL=ON \
  -D TPL_ENABLE_Pnetcdf:BOOL=ON \
  -D TPL_ENABLE_HDF5=ON \
  -D EpetraExt_ENABLE_HDF5=OFF \
  -D HDF5_INCLUDE_DIRS=/usr/local/Cellar/hdf5-mpi/1.14.3/include \
  -D HDF5_LIBRARY_DIRS=/usr/local/Cellar/hdf5-mpi/1.14.3/lib \
  -D Netcdf_INCLUDE_DIRS=/usr/local/include \
  -D Netcdf_LIBRARY_DIRS=/usr/local/lib \
  -D TPL_ENABLE_Matio=OFF \
  -D TPL_ENABLE_X11=OFF \
  -D Trilinos_ENABLE_Stokhos:BOOL=OFF \
  -D Trilinos_ENABLE_Fortran:BOOL=ON \
  -D Trilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=ON \
```

**Comment by dridzal** [comment]
After some bisection

- master fails to link
- This recent SHA1 builds correctly: bd79a63bbb81c37a0b60dd5468b2078ac37f96e2
    https://github.com/trilinos/Trilinos/pull/13046

Please advise.

**Comment by brian-kelley** [comment]
@dridzal Me and @cgcgcg looked at this. @cgcgcg suggested that this function just needs to be added to ETI. I replicated the error and checked that that change fixes it: #13109

**Comment by brian-kelley** [comment]
@dridzal This should be fixed now in develop, can you try the build again?
